### PR TITLE
Add identities GC test based on control-plane framework

### PR DESCRIPTION
**Pull request by pippolo84**
This PR leverages the control-plane test framework to add a new unit test equivalent to the `test/k8s/identities.go` Ginkgo test.

The test will verify that, after the amount of time specified in the operator config, the "dummy identity" will be garbage collected.

The operator code has been modified to run in memory and to be able to be reset and restarted again in the same test binary. This way, the framework is able to support future operator-related test cases without any further change.

### CODE REVIEW

File: test/controlplane/suite/testcase.go [hunk 1]
```diff
@@ -79,11 +91,48 @@ func NewControlPlaneTest(t *testing.T, nodeName string, k8sVersion string) *Cont
 	}
 }
 
-func (cpt *ControlPlaneTest) StartAgent(modConfig func(*option.DaemonConfig)) *ControlPlaneTest {
+func (cpt *ControlPlaneTest) SetupEnvironment(modConfig func(*agentOption.DaemonConfig, *operatorOption.OperatorConfig)) *ControlPlaneTest {
```
Comment (ldelossa) [inline test/controlplane/suite/testcase.go:97]: Probably makes sense to comment this.

Comment (ldelossa) [inline test/controlplane/suite/testcase.go:97]: Why is this new method strictly necessary in the PR? 

Would it be possible to simply use the api:

```
StartAgent(func(*agentOption.DaemonConfig))
StartOperator(func(*operatorOption.OperatorConfig))
````

Where each function sets the default config plus any additional passed in and starts each of their components. 

This makes it such that all tests do not need to change.

**Review by ldelossa — CHANGES_REQUESTED** [review]

File: test/controlplane/suite/testcase.go (hunk 1 above)
Comment (pippolo84) [reply test/controlplane/suite/testcase.go:97]: In `SetupEnvironments` there are some operations that affect the entire test environment, such the k8s clients and the DNS proxy initialization. These operations should be done once and in all possible setups (agent-only test, operator-only test, agent+operator test) so I think that a separate "setup" api will be more coherent for a consumer of the framework.

Besides, some "agent options" are really `config.Option` singleton values and will affect the behavior of the operator, too. IOW, with the previous api, calling or not calling `StartAgent` may affect an operator-only test too, IMHO leading to possible confusion.

Comment (ldelossa) [reply test/controlplane/suite/testcase.go:97]: Okay this makes sense to me.

**Review by ldelossa — APPROVED** [review]

File: operator/cmd/root.go [hunk 2]
```diff
@@ -605,3 +605,9 @@ func OnOperatorStartLeading(ctx context.Context) {
 
 	log.Info("Initialization complete")
 }
+
+// ResetCiliumNodesCacheSyncedStatus resets the current status of
+// cache synchronization in Cilium nodes as "not synced".
```
Comment (aanm) [inline operator/cmd/root.go:610]: Might be worth adding a comment saying that this should only be used in tests.

**Review by aanm — APPROVED** [review]
LGTM only one small non-blocking comment.

File: operator/cmd/root.go (hunk 2 above)
Comment (ldelossa) [reply operator/cmd/root.go:610]: Agree with this, at first I didn't like that a method floated up to the runtime code that is only used in a test, but after seeing that its just a function defined in the `cmd` package, I felt OK with it. That being said, commenting would be a great idea.

Comment (pippolo84) [reply operator/cmd/root.go:610]: I'll add a comment. Thank you for the review.

**Review by joamaki — APPROVED** [review]
LGTM

**Pull request merged**